File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] nums) {
        int l=0;
        int r=nums.length-1;
        int maxa=0;
        while(l<=r){
            int b=r-l;
            int area=Math.min(nums[l],nums[r])*b;
            maxa=Math.max(maxa,area);
            if(nums[l]<=nums[r]){
                l++;
            }else{
                r--;
            }

        }return maxa;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
    
        i = 0
        j = len(height)-1
        max = 0
        while i != j:
            x = j - i
            if height[i] <= height[j]:
                area = height[i] * x
                i += 1
            else:
                area = height[j] * x
                j -= 1
            if area > max:
                max = area
        return max